crypto/x509/algorithm.c

```diff
@@ -120,8 +120,13 @@ int x509_digest_verify_init(EVP_MD_CTX *ctx, X509_ALGOR *sigalg,
     return 0;
   }
 
-  /* Check the public key OID matches the public key type. */
-  if (pkey_nid != EVP_PKEY_id(pkey)) {
+  /* Check the public key OID matches the public key type.
+   * One special case is |EVP_PKEY_id(pkey)| may be |EVP_PKEY_RSA_PSS|.
```

Comment:
Not clear to me why this special case happens? Can we fix that?

Reply:
This special case happens because `NID_rsassaPss` of `sigalg_nid` can be used when the key type is `EVP_PKEY_RSA` or `EVP_PKEY_RSA_PSS`.  The later one is PSS only key.

> Can we fix that?

Can you specify the fix? To remove the special case, we may need to rewrite the mapping logics of `OBJ_find_sigid_algs` and consider [other special cases](https://github.com/awslabs/aws-lc/blob/main/crypto/obj/obj_xref.c#L88-L107).  Additional evaluation is needed to avoid side-effects.